web_poet/rules.py

```diff
@@ -81,52 +81,51 @@ def __hash__(self):
         return hash((self.for_patterns, self.use, self.instead_of, self.to_return))
 
 
-class PageObjectRegistry(dict):
-    """This contains the :class:`~.ApplyRule` that associates the Page Objects
-    alongside its Items for a given URL matching rule.
-
-    PageObjectRegistry is a ``dict`` subclass with added functionalities on
-    storing, retrieving, and searching for the :class:`~.ApplyRule` instances.
-    The **value** represents the :class:`~.ApplyRule` instance from which the
-    Page Object in the **key** is allowed to be used. Since it's essentially a
-    ``dict``, you can use any ``dict`` operations with it.
+class RulesRegistry(dict):
```

Comment:
Related to https://github.com/scrapinghub/web-poet/issues/86#issuecomment-1297274048.

What do you think about retaining `PageObjectRegistry` for backwards compatibility with the `dict`-interface, but then introducing a new Registry that's not based on `dict`? In this way, we can support multiple rules for a single Page Object. 

I imagine that we need this before the **0.6.0** release since it's a blocker for use cases like when users want to use the same Page Object that supports different `to_return` values.

Reply:
> What do you think about retaining PageObjectRegistry for backwards compatibility with the dict-interface, but then introducing a new Registry that's not based on dict? In this way, we can support multiple rules for a single Page Object.

I might be missing something, but for me it looks like the direct use of PageObjectRegistry is next to non-existent now, and so we can be pretty agressive deprecating and removing features. It'd be a larger issue after handle_urls cleanup is released, together with scrapy-poet changes, as I'd expect the usage to increase.

> I could see how we need this before the 0.6.0 release

I think cleaning up API is a blocker, but supporting multiple rules is not a blocker. We can add this support later, and there's enough useful code to release in web-poet.

Reply:
Even if we don't go with #86, it still can be a good idea to make the exposed API smaller, as it allows us to make changes easier without breaking user code.

Reply:
This may be the cleanest, otherwise we would need to reimplement all methods of dict to mark them as deprecated, and I wonder if some of the [pitfalls of dict-inheritance](https://realpython.com/inherit-python-dict/#inheriting-from-the-python-built-in-dict-class) could be an issue for that.

Reply:
Ok, let me give it a try

Reply:
Done!